pallets/connectors-gateway/connectors-gateway-axelar-precompile/src/lib.rs

```diff
@@ -31,13 +32,63 @@ pub type Bytes<const U32: u32> = BoundedBytes<ConstU32<U32>>;
 
 pub use pallet::*;
 
+#[derive(
+	PartialEq,
+	Clone,
+	codec::Encode,
+	codec::Decode,
+	scale_info::TypeInfo,
+	codec::MaxEncodedLen,
+	frame_support::RuntimeDebugNoBound,
+)]
+pub struct SourceConverter {
+	domain: Domain,
+}
+
+impl SourceConverter {
+	pub fn try_convert(&self, maybe_address: &[u8]) -> Option<DomainAddress> {
+		match self.domain {
+			Domain::Centrifuge => Some(DomainAddress::Centrifuge(Self::try_into_32bytes(
+				&maybe_address,
+			)?)),
+			Domain::EVM(id) => Some(DomainAddress::EVM(
+				id,
+				Self::try_into_20bytes(&maybe_address)?,
+			)),
+		}
+	}
```

Comment:
This makes sense! Can't we implement `TryInto` or `TryFrom` instead tho?

Reply:
Both of these take only into account `self` but we need an external input. Like `TryConvert` but this is not available on our Subsrtate version yet. Or am I missing some>

Reply:
We could always include that as having the conversion being between a (Domain, Address) and DomainAddress but this is fine 👍🏻